### PR TITLE
Add CIS Kubernetes Foundations Benchmark v1.8.0-(10-05-2023) closes #76

### LINKED ISSUE #76: Add CIS Kubernetes Foundations Benchmark v1.8.0-(10-05-2023)
**Is your feature request related to a problem? Please describe.**
CIS [Reference](https://www.cisecurity.org/benchmark/kubernetes)
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: cis_v180/section_2.sp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  tags = merge(local.cis_v180_2_common_tags, {
+    cis_level   = "1"
+    cis_item_id = "2.1"
+    cis_type    = "automated"
+    type        = "Benchmark"
+  })
+}
+
+benchmark "cis_v180_2_2" {
+  title         = "2.2 Ensure that the --client-cert-auth argument is set to true "
```
Comment (madhushreeray30) [inline cis_v180/section_2.sp:47]:
```suggestion
  title         = "2.2 Ensure that the --client-cert-auth argument is set to true"
```

File: cis_v180/section_2.sp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  tags = merge(local.cis_v180_2_common_tags, {
+    cis_level   = "1"
+    cis_item_id = "2.4"
+    cis_type    = "automated"
+    type        = "Benchmark"
+  })
+}
+
+benchmark "cis_v180_2_5" {
+  title         = "2.5 Ensure that the --peer-client-cert-auth argument is set to true "
```
Comment (madhushreeray30) [inline cis_v180/section_2.sp:119]:
```suggestion
  title         = "2.5 Ensure that the --peer-client-cert-auth argument is set to true"
```

File: cis_v180/section_4.sp [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  tags = merge(local.cis_v180_4_2_common_tags, {
+    cis_level   = "1"
+    cis_item_id = "4.2.6"
+    cis_type    = "automated"
+    type        = "Benchmark"
+  })
+}
+
+benchmark "cis_v180_4_2_7" {
+  title         = "4.2.7 Ensure that the --hostname-override argument is not set "
```
Comment (madhushreeray30) [inline cis_v180/section_4.sp:189]:
```suggestion
  title         = "4.2.7 Ensure that the --hostname-override argument is not set"
```

File: cis_v180/section_4.sp [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    cis_level   = "1"
+    cis_item_id = "4.2.9"
+    cis_type    = "manual"
+    type        = "Benchmark"
+  })
+}
+
+benchmark "cis_v180_4_2_10" {
+  title         = "4.2.10 Ensure that the --tls-cert-file and --tls-private-key-file arguments are set as appropriate"
+  description   = "Setup TLS connection on the Kubelets."
```
Comment (madhushreeray30) [inline cis_v180/section_4.sp:237]:
```suggestion
  description   = "Enable kubelet client certificate rotation."
```

File: cis_v180/section_4.sp [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+  tags = merge(local.cis_v180_4_2_common_tags, {
+    cis_level   = "1"
+    cis_item_id = "4.2.9"
+    cis_type    = "manual"
+    type        = "Benchmark"
+  })
+}
+
+benchmark "cis_v180_4_2_10" {
+  title         = "4.2.10 Ensure that the --tls-cert-file and --tls-private-key-file arguments are set as appropriate"
```
Comment (madhushreeray30) [inline cis_v180/section_4.sp:236]:
```suggestion
  title         = "4.2.10 Ensure that the --rotate-certificates argument is not set to false"
```

**Review by madhushreeray30 — COMMENTED** [review]
Please take a look at the comments, thanks!

File: cis_v180/section_1.sp [hunk 6]
```diff
@@ -0,0 +1,843 @@
...
+    benchmark.cis_v180_1_2_28,
+    benchmark.cis_v180_1_2_30
+  ]
+
+  tags = merge(local.cis_v180_1_common_tags, {
+    type = "Benchmark"
+  })
+}
+
+benchmark "cis_v180_1_2_2" {
```
Comment (rajlearner17) [inline cis_v180/section_1.sp:71]: @khushboo9024 As discussed, we may have to generate specific queries to check only the required CIS parameters, e.g. for `pod_container_token_auth_file_not_configured`, in this one, we need to validate the container name is like `%kube-apiserver%`, then validate the conditions further else `skip`

Comment (rajlearner17) [reply cis_v180/section_1.sp:71]: We may need to apply the same logic across all queries wherever such specifications for control plane components such as `kube-controller-manager`, 'etcd' etc...

**Comment by khushboo9024** [comment]
Currently, this isn't needed as we have plans to add provider-specific CIS benchmarks and controls. (Example: CIS Google Kubernetes Engine (GKE) Benchmark)

**Pull request closed without merging**